File: 9월 13일 - 정수론/2108.cpp
// 정수론: 2108 통계학
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>
using namespace std;

int mean(int n, vector<int> &v){
    int sum = 0;
    for (int i=0; i<n; i++){
        sum += v[i];
    }
    return round(float(sum)/n);
}

int mid(int n, vector<int> &v){
    sort(v.begin(), v.end()); // 오름차순 정렬
    return v[floor(n/2)]; // n은 홀수
}

int mode(int n, vector<int> &v){
    bool not_first = false;
    int mode_num = -4001;
    vector<int> cnt (8001, 0); // -4000 ~ 4000
    for (int i=0; i<n; i++){ // 빈도수 저장
        cnt[v[i] + 4000]++;
    }
    int max = *max_element(cnt.begin(), cnt.end());
    for (int i=0; i<8001; i++){
        if (cnt[i] != max){
            continue;
        } else{
            if (!not_first){
                not_first = true;
                mode_num = i-4000;
            } else {
                return i-4000;
            }
        }
    }
    return mode_num;
}

int range(vector<int> &v){
    sort(v.begin(), v.end()); 
    return v.back() - v.front();
}

int main(){
    int n;
    cin >> n;
    vector<int> num (n, 0);
    for (int i=0; i<n; i++){
        cin >> num[i];
    }
    cout << mean(n, num) << "\n" << mid(n, num) << "\n" << mode(n, num) << "\n" << range(num);
    return 0;
}

File: 9월 13일 - 정수론/Python/2108.py
# 정수론: 2108 통계학
import sys, statistics
from collections import Counter
def getMode(arr):
    cnt = Counter(arr)
    most_cnt = cnt.most_common(2)
    if (len(cnt) > 1):
        if most_cnt[0][1] == most_cnt[1][1]:
            return most_cnt[1][0]
        else:
            return most_cnt[0][0]
    else:
        return most_cnt[0][0]
input = sys.stdin.readline
N = int(input())
arr = []
for i in range(N):
    num = int(input())
    arr.append(num)
arr.sort()
print(round(statistics.mean(arr)))
print(statistics.median(arr))
print(getMode(arr))
print(arr[-1] - arr[0])
"""
def getMode(arr):
    cnt = Counter(arr)
    most_cnt = cnt.most_common()
    most = 0
    most = most_cnt[0][1]
    most_num = []
    for i in most_cnt:
        if i[1] == most:
            most_num.append(i[0])
    if len(most_num) == 1:
        return most_num[0]
    else:
        most_num.sort()
        return most_num[1]
"""